src/perception/PoseEstimatorModule.cpp

```diff
@@ -91,7 +98,30 @@ bool PoseEstimatorModule::detectObjects(
     ros::Time t0 = ros::Time(0);
 
     // get the object name, resource, and offset from database by objectKey
-    mConfigData->getObjectByKey(obj_key, obj_name, obj_resource, obj_offset);
+    try {
+      mConfigData->getObjectByKey(obj_db_key, obj_name, obj_resource, obj_offset);
+    } catch(std::runtime_error &e) {
+      dtwarn << e.what() << std::endl;
+      continue;
+    }
+
+    // Add object to output vector, if available
+    if(detectedObjects != nullptr) {
+      detectedObjects->push_back(this_object);
```

Comment:
Is there a reason to construct `DetectedObject` and then `push_back`, rather than `emplace_back`? It seems like we have access to all the strings already anyway.

Reply:
The constructed object is used later in the function, so I don't believe there is a penalty to using `push_back`.